pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -305,3 +306,82 @@ func TestLookupIPFallback(t *testing.T) {
 	err := f.lookupIP(ctx, "www.google.com")
 	require.NoError(t, err, "Error when resolving name")
 }
+
+func TestString(t *testing.T) {
+	tests := []struct {
+		name           string
+		selectorItem   *fqdnSelectorItem
+		expectedOutput string
+	}{
+		{
+			name: "matching the regex",
+			selectorItem: &fqdnSelectorItem{
+				matchRegex: "^.*antrea[.]io$",
+			},
+			expectedOutput: "matchRegex:^.*antrea[.]io$",
+		},
+		{
+			name: "matching the name",
+			selectorItem: &fqdnSelectorItem{
+				matchName: "test.antrea.io",
+			},
+			expectedOutput: "matchName:test.antrea.io",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			gotOutput := tc.selectorItem.String()
+			assert.Equal(t, tc.expectedOutput, gotOutput)
+		})
+	}
+}
+
+func TestGetIPsForFQDNSelectors(t *testing.T) {
+	selectorItem := fqdnSelectorItem{
+		matchName: "test.antrea.io",
+	}
+	tests := []struct {
+		name                       string
+		fqdns                      []string
+		existingSelectorItemToFQDN map[fqdnSelectorItem]sets.String
+		existingDNSCache           map[string]dnsMeta
+		expectedMatchedIPs         []net.IP
+	}{
+		{
+			name:  "matched ip found",
+			fqdns: []string{"test.antrea.io"},
+			existingSelectorItemToFQDN: map[fqdnSelectorItem]sets.String{
+				selectorItem: sets.NewString("test.antrea.io"),
+			},
+			existingDNSCache: map[string]dnsMeta{
+				"test.antrea.io": {
+					responseIPs: map[string]net.IP{
+						"IP": net.ParseIP("127.0.0.1"),
```

Comment:
The key doesn't match the data in real world. It should be the string format of the IP.
And please use more than one IP to cover multiple response IPs case.

Reply:
Done